api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -35,6 +39,62 @@ public String readMandrillApiKey() {
     return mandrillKeys.getString("api-key");
   }
 
+  public void copyAllDemoNotebooks(String workspaceBucket)  {
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    Bucket demoBucket = storage.get(getDemosBucketName());
+    Page<Blob> allBlobs = demoBucket.list();
+    List<Blob> notebookFiles = new ArrayList<>();
+    notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    while (allBlobs.hasNextPage()) {
+      allBlobs = allBlobs.getNextPage();
+      notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    }
```

Comment:
This whole block feels like it could be much simpler in a stream. I haven't delved into it in the IDE, but will give that a shot.